File: app/java/src/main/java/com/example/production/domain/model/inventory/IssueInstruction.java
package com.example.production.domain.model.inventory;

import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 払出指示データエンティティ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class IssueInstruction {
    private Long id;
    private String instructionNumber;
    private String orderNumber;
    private LocalDate instructionDate;
    private String locationCode;
    private String remarks;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private List<IssueInstructionDetail> details;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Inventory/IssueInstruction.cs
namespace ProductionManagement.Domain.Models.Inventory;

/// <summary>
/// 払出指示
/// </summary>
public class IssueInstruction
{
    public long? Id { get; init; }

    public required string InstructionNumber { get; init; }

    public required string OrderNumber { get; init; }

    public required DateOnly InstructionDate { get; init; }

    public required string LocationCode { get; init; }

    public string? Remarks { get; init; }

    public DateTime CreatedAt { get; init; }

    public DateTime UpdatedAt { get; init; }

    public IReadOnlyList<IssueInstructionDetail>? Details { get; set; }
}
